scripts/utils.sh

```
@@ -314,6 +314,27 @@ function clean_resources() {
+        log "ERROR" "Could not find /etc/machine-id file."
+        exit 1
+    fi
+
+    local machine_id
+    read -r machine_id < /etc/machine-id
+    local combined="${machine_id}-${vm_name}"
+    local hash
+    hash=$(printf "%s" "$combined" | sha256sum | cut -c1-10)
+
+    # Use QEMU's standard locally administered MAC prefix (52:54:00)
+    local mac="52:54:00:$(echo "$hash" | sed 's/\(..\)\(..\)\(..\).*/\1:\2:\3/')"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)